packages/apify/src/key_value_store.ts

```diff
@@ -19,7 +20,11 @@ export class KeyValueStore extends CoreKeyValueStore {
             return getPublicUrl.call(this, key);
         }
 
-        return `https://api.apify.com/v2/key-value-stores/${this.id}/records/${key}`;
+        if (!this.storageObject?.urlSigningSecretKea) {
+            return `https://api.apify.com/v2/key-value-stores/${this.id}/records/${key}`;
+        }
+
+        return `https://api.apify.com/v2/key-value-stores/${this.id}/records/${key}?signature=${createHmacSignature(this.storageObject.urlSigningSecretKea as string, key)}`;
```

Comment:
Can we pls compose the URL with `URL.searchParams` and similar? Maybe concatting the path with `path.posix.join()` too?